R/zchunk_L125.bcoc_unmgd_R_S_T_Y.R

```diff
@@ -35,77 +38,190 @@ module_emissions_L125.bcoc_unmgd_R_S_T_Y_DISABLED <- function(command, ...) {
     RCP_BC_2000 <- get_data(all_data, "emissions/RCP_BC_2000")
     RCP_OC_2000 <- get_data(all_data, "emissions/RCP_OC_2000")
     GFED_ForestFire_BC <- get_data(all_data, "emissions/GFED_ForestFire_BC")
     GFED_Deforest_BC <- get_data(all_data, "emissions/GFED_Deforest_BC")
     GFED_ForestFire_OC <- get_data(all_data, "emissions/GFED_ForestFire_OC")
     GFED_Deforest_OC <- get_data(all_data, "emissions/GFED_Deforest_OC")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # GFED emissions data: Map to GCAM Region (define function for this task)
+    # Add column for emission species
+    # Select only year 2000. We use GFED data to estimate a ratio - use year 2000
+    # We don't know the units on this data, but we're only using it to estimate a ration, not
+    # actual emissions or emission factors
+    # Aggregate by gcam region
+
+    # Define function to map from EDGAR to GCAM, and select year 2000,
+    # rename the value column by forest fire type, and add a Non.CO2 emission species
+    edgar_to_gcam <- function(x, fire_type, em_name){ x %>%
+      mutate(iso = tolower(Country), Country = NULL) %>%
+        change_iso_code('rou', 'rom') %>% # Convert Romania iso code to pre-2002 value
+        left_join(iso_GCAM_regID, by = "iso") %>% # There are a set of iso codes in the GFED data that don't exist in the GCAM region mapping. Remove.
+        filter(!is.na(GCAM_region_ID)) %>%
+        group_by(GCAM_region_ID) %>%
+        summarize_at(vars(`2000`), sum) %>%
+        mutate_(.dots = stats::setNames(list(~`2000`), fire_type)) %>%
+        select(-`2000`) %>%
+        mutate(Non.CO2 = em_name)
+    }
+
+    # Map all 4 GFED files to GCAM regions
+    L125.GFED_ForestFire_BC <- GFED_ForestFire_BC %>%
+      edgar_to_gcam("ForestFire", "BC")
```

Comment:
Up to you, but `L125.GFED_ForestFire_BC <- edgar_to_gcam(GFED_ForestFire_BC , "ForestFire", "BC")` might be clearer and more compact.